public/app/core/components/Login/LoginForm.tsx

```diff
@@ -0,0 +1,123 @@
+  private userInput: HTMLInputElement;
+  constructor(props: Props) {
+    super(props);
+    this.state = {
+      user: '',
+      password: '',
+      email: '',
+      valid: false,
+    };
+  }
+
+  componentDidMount() {
+    this.userInput.focus();
+  }
+  onSubmit = (e: SyntheticEvent) => {
+    e.preventDefault();
+
+    const { user, password, email } = this.state;
+    if (this.state.valid) {
+      this.props.onSubmit({ user, password, email }, this.state.valid);
+    }
+  };
+
+  onChange = (e: SyntheticEvent) => {
+    // @ts-ignore
+    this.setState({ [e.target.name]: e.target.value }, () => {
+      this.setState({ valid: this.validate() }, () => {
+        if (this.props.onChange) {
+          this.props.onChange(this.state.valid);
+        }
+      });
+    });
+  };
+
+  validate() {
+    if (this.state.user.length > 0 && this.state.password.length > 0) {
+      return true;
+    } else {
+      return false;
+    }
+  }
+
+  render() {
+    return this.props.displayLoginFields ? (
+      <form name="loginForm" className="login-form-group gf-form-group">
+        <div className="login-form">
+          <input
+            ref={input => {
+              this.userInput = input;
+            }}
+            type="text"
+            name="user"
+            className="gf-form-input login-form-input"
+            required
+            // ng-model="formModel.user"
+            placeholder={this.props.loginHint}
+            aria-label="Username input field"
+            onChange={this.onChange}
+          />
+        </div>
+        <div className="login-form">
+          <input
+            type="password"
+            name="password"
+            className="gf-form-input login-form-input"
+            required
+            ng-model="formModel.password"
+            id="inputPassword"
+            placeholder={this.props.passwordHint}
+            aria-label="Password input field"
+            onChange={this.onChange}
+          />
+        </div>
+        <div className="login-button-group">
+          {!this.props.isLoggingIn ? (
+            <button
+              type="submit"
+              aria-label="Login button"
+              className={`btn btn-large p-x-2 ${this.state.valid ? 'btn-primary' : 'btn-inverse'}`}
+              onClick={this.onSubmit}
+              disabled={!this.state.valid}
+            >
+              Log In
+            </button>
+          ) : (
+            <button type="submit" className="btn btn-large p-x-2 btn-inverse btn-loading">
+              Logging In<span>.</span>
+              <span>.</span>
+              <span>.</span>
+            </button>
+          )}
+
+          {this.props.displayForgotPassword ? (
+            <div className="small login-button-forgot-password">
+              <a href="user/password/send-reset-email">Forgot your password?</a>
+            </div>
+          ) : null}
+        </div>
+      </form>
+    ) : null;
```

Comment:
Why are you rendering nothing here? You could decide whether or not to render the LoginForm in the LoginPage component, and keep the visibility information in the LoginCtrl